### PR TITLE
fix: tolerate partial property/post payloads from any device

### PR DESCRIPTION
**Pull request by jbowen116**
## Summary

Mammotion devices routinely send `thing.event.property.post` messages containing only a small subset of fields. A Spino E1 debug capture in https://github.com/mikey0000/Mammotion-HA/issues/763 showed posts with as few as 2 fields:

```
{"params":{"intMod":"SPINO E1","extMod":"SPINO E1"}, ...}
{"params":{"deviceVersion":"1.15.2.1039","deviceVersionInfo":"..."}, ...}
```

But `DeviceProperties` marked nearly every field as required, so `mashumaro` raised `MissingField` on the first absent field and every partial post was silently dropped at `_dispatch_mammotion_properties`:

```
mashumaro.exceptions.MissingField: Field "device_state" of type int is missing in DeviceProperties instance
```

Same family as #153 and #158, broader scope (any device, any partial post).

## What changes

**`DeviceProperties`** (`pymammotion/data/mqtt/mammotion_properties.py`):
- Every scalar field gets a sensible default (`0` for ints, `""` for strs).
- Every nested object becomes `T | None = None` — the post-omitted JSON-string lambdas in `serialization_strategy` already pass `None` through unchanged.

**`MowerStateReducer.apply_mammotion_properties`** (`pymammotion/device/state_reducer.py`):
- Add explicit `is not None` guards around `device_version_info` access (the existing try/except still catches AttributeError; this is for clarity).
- **Fix a pre-existing indentation bug**: the `coordinate` update block sat inside the `device_version_info` `except` handler, so it only ran when the firmware-info parse failed. Dedented to run unconditionally with its own `is not None` guard.
- Wrap the `network_info` body in `if net is not None:` so the AttributeError fall-through is no longer load-bearing.

## Verification

Confirmed end-to-end against:
- **Real Spino E1 partial payloads** from #763 (both the model-only and firmware-only variants) — both now parse cleanly with sensible defaults for absent fields.
- **Existing Yuka Mini 2 fixture** (`tests/fixtures/yuka_mini2_property_post.json`) — full-payload parsing remains unchanged; `battery_percentage=31`, `device_state=13`, `network_info.wifi_rssi=-65`, `device_other_info.iot_con_fail_min=0` all populate correctly.

New regression test in `tests/test_mammotion_properties_partial_post.py` covers both Spino variants plus an empty-params edge case.

## Scope notes

This does **not** address the headline #763 complaint (`sys_status` stuck at `PREPARE` while the Spino is supposedly cleaning). That state field comes from the protobuf `DevStatueT` path, not the JSON property/post — and the user's log capture is too short and at full battery to confirm the reporter is actually mid-clean. A longer capture during active cleaning is still needed to triage that.

## Test plan

- [ ] CI passes (the 18 pre-existing failures in `tests/messaging/test_hash_list_geojson.py` + `tests/render/test_svg_center.py` also fail on `main` and are unrelated)
- [ ] `pytest tests/test_mammotion_properties_partial_post.py tests/test_mammotion_properties_yuka_mini2.py tests/test_network_info_luba2_awd.py` — all pass locally
- [ ] `ruff format --check` clean on the touched files

### CODE REVIEW

File: pymammotion/device/state_reducer.py [hunk 1]
```diff
@@ -770,32 +767,35 @@ def apply_mammotion_properties(  # type: ignore
                 _apply_mower_fw_module(device.device_firmwares, fw_type, v)
 
         try:
-            if p.device_version_info.dev_ver:
-                device.device_firmwares.device_version = p.device_version_info.dev_ver
-            for module in p.device_version_info.fw_info:
-                if module.v:
-                    _apply_mower_fw_module(device.device_firmwares, module.t, module.v)
+            if p.device_version_info is not None:
```
Comment (mikey0000) [inline pymammotion/device/state_reducer.py]: if device_version_info := p.device_version_info:

Is much cleaner

**Comment by mikey0000** [comment]
Just want to make known that I really appreciate the extra help

**Review by mikey0000 — APPROVED** [review]
